src/core/utils/transactions.ts

```diff
@@ -278,19 +281,23 @@ export function parseTransaction({
   const description = getDescription(asset, type, meta);
 
   const nativeAsset = changes.find((change) => change?.asset.isNativeAsset);
-  const nativeAssetPrice = FixedNumber.fromString(
-    nativeAsset?.price?.toString() || '0',
-  );
 
   const value = FixedNumber.fromValue(
     BigNumber.from(nativeAsset?.value || 0),
     nativeAsset?.asset.decimals || 0,
   );
 
-  const valueInNative = value.mulUnsafe(nativeAssetPrice).toString();
-
   const nativeAssetDecimals = 18; // we only support networks with 18 decimals native assets rn, backend will change when we support more
 
+  const nativeAssetPrice = FixedNumber.fromString(
+    typeof nativeAsset?.price === 'number'
```

Comment:
in what case is it not a number? can we maybe assert this some other way?

Reply:
how would u suggest this be done

i just wanna check that it exists and still handle 0

Reply:
wouldn't this work?
```ts
const nativeAssetPrice = FixedNumber.fromString(
    nativeAsset?.price ? nativeAsset.price.toFixed(nativeAssetDecimals).toString() : '0',
    nativeAssetDecimals,
  );
```

Reply:
This does look cleaner. I guess I prefer being more defensive since we're calling `toFixed` and this would blow up if we were accidentally sent a string.